### PR TITLE
Add doc requirements to PR template

### PR DESCRIPTION
**Pull request by meker12**
# Description

Add release note and documentation requirements to PR template

### Fixed Issues (if relevant)

n/a

### Manual testing scenarios
n/a

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -30,8 +30,17 @@
...
+ If your proposed update requires user documentation, submit a PR to the Magento DevDocs repository. For extensive updates requiring assistance, submit an issue to DevDocs. See https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md.
+ -->
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
+ - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
  - [ ] All commits are accompanied by meaningful commit messages
- - [ ] All new or changed code is covered with unit/integration tests (if applicable)
- - [ ] All automated tests passed successfully (all builds on Travis CI are green)
```
Comment (NadiyaS) [inline .github/PULL_REQUEST_TEMPLATE.md:37]: Why we removed these 2 items?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -30,8 +30,17 @@
...
+For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud ECE-tools release notes](https://devdocs.magento.com/cloud/release-notes/ece-release-notes.html).
+
+### Associated documentation updates
+<!--
+ If your proposed update requires user documentation, submit a PR to the Magento DevDocs repository. For extensive updates requiring assistance, submit an issue to DevDocs. See https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md.
+ -->
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
```
Comment (BarnyShergold) [inline .github/PULL_REQUEST_TEMPLATE.md:46]: Please can we have this clarified that it is not sufficient to just put a link to a JIRA ticket as the description? This is often done and the issue is that Community Maintainers like myself do not have access to these tickets.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (meker12) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: Restored these items. This was a copy/paste error.  The PR templates in the Cloud docker, patches, and components repositories do not have these items in the checklist.  Do they need to be added to those PR templates?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (meker12) [reply .github/PULL_REQUEST_TEMPLATE.md:46]: I added clarification in the Description comments.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -30,8 +32,17 @@
...
+ - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
```
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (NadiyaS) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: It's OK not to have it for docker PRs and MCC. But steps for manual tests should be required

**Review by NadiyaS — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by shiftedreality** [comment]
@BarnyShergold, please approve

**Review by BarnyShergold — APPROVED** [review]

**Pull request merged**